app/src/main/java/com/example/tasky/actions/presentation/ActionViewState.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.tasky.actions.presentation
+
+data class ActionViewState(
+    val actionTitle: String = "",
+    val actionDescription: String = "",
+    val startTime: String = "",
+    val endTime: String = "",
+    val startDate: String = "",
+    val endDate: String = "",
+    val reminder: Int = 0,
+    val actionViewMode: ActionViewMode = ActionViewMode.EDITABLE,
+    val attendeeFilterSelected: AttendeeFilter = AttendeeFilter.ALL
+)
+
+enum class AttendeeFilter(type: String) {
+    ALL("All"), GOING("Going"), NOT_GOING("Not Going")
+}
+
+enum class ActionViewMode {
+    NON_EDITABLE, EDITABLE, ATTENDEE, ATTENDEE_EDITABLE
+}
```

Comment:
There are actually two distinct states hidden here which are not mutually exclusive (you can be both an attendee and it's editable). You've encoded that with two extra enums, but an enum is not the right data structure in that case, since it models mutually exclusive things by default.
If something is not mutually exclusive, better work with booleans:
```
val isEditable: Boolean = false
val isLocalUserAttendee: Boolean = false
```